### PR TITLE
implement active/inactive state restore

### PR DESCRIPTION
**Pull request by neffo**
See #8 

Added a third setting for the indicator toggle which will restore the video mode after a shell or system restart. This is indicated by a small lock on the icon.

I've added an additional dconf key to store this. I've tried to minimise other code changes.

### CODE REVIEW

**Review by cyberalex4life — APPROVED** [review]

**Comment by cyberalex4life** [comment]
@neffo and @jenspfahl , I think there's a misspelled variable on this commit:
"extension.js", line 402
`_mode = _MODE_ON_LOCK; `
should be
`_mode = MODE_ON_LOCK;`
I get _MODE_ON_LOCK  undefined variable.

**Comment by neffo** [comment]
@cyberalex4life fixed!

File: KeepAwake@jepfa.de/extension.js [hunk 1]
```diff
@@ -91,6 +95,14 @@ function setPowerAc(value) {
...
 }
 
+function getStateRestore() {
+    return _extensionSettings.get_boolean(RESTORE_STATE_KEY);
+}
+
+function setStateRestore(value) {
+    _extensionSettings.set_boolean(RESTORE_STATE_KEY, value);
+}
+
```
Comment (jenspfahl) [inline KeepAwake@jepfa.de/extension.js]: Please move both methods below getPowerBat(), so every property-getter/setter are side by side.

File: KeepAwake@jepfa.de/extension.js [hunk 2]
```diff
@@ -160,18 +172,26 @@ function isReadyForWatchingVideo() {
...
-        // on --> off
+    if (_mode == MODE_ON) {    
+
+        // on --> on with lock
+	    _mode = MODE_ON_LOCK;
+	    setStateRestore(true);            
+    } 
+    else if (_mode == MODE_ON_LOCK) {
+    
+        // on --> on with lock
```
Comment (jenspfahl) [inline KeepAwake@jepfa.de/extension.js]: Comment should be `// on with lock --> off`

File: KeepAwake@jepfa.de/extension.js [hunk 3]
```diff
@@ -200,7 +220,7 @@ function updateMode() {
 function showModeTween() {
 
   
-    if (_mode == MODE_ON) {
+    if (_mode == MODE_ON || _mode == MODE_ON_LOCK) {
         _tweenText = new St.Label({ style_class: 'video-label-on', text: _("Computer keeps awake.") });
```
Comment (jenspfahl) [inline KeepAwake@jepfa.de/extension.js]: I think it is better to show an additional message when toggling to MODE_ON_LOCK, e.g. "Computer keeps awake even after restart. "

**Review by jenspfahl — CHANGES_REQUESTED** [review]
Would be nice to shortly mention the new feature in README.md. Please see my other comments.

Btw, good work!

**Comment by neffo** [comment]
Sorry, not sure how to add the README.md file to this pull request, but other changes have been made as requested.

EDIT: nevermind, worked that out (read the instructions)! All done.

**Review by jenspfahl — APPROVED** [review]

**Pull request merged**